packages/reapit-connect-management/src/components/ui/users/edit-user-group.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react'
+import useSWR from 'swr'
+import difference from 'lodash.difference'
+import { FormFieldInfo } from '@reapit/utils'
+import {
+  Button,
+  Section,
+  ModalV2,
+  Formik,
+  Form,
+  DropdownSelect,
+  Loader,
+  SelectOption,
+  notification,
+} from '@reapit/elements'
+import { GroupModel, UserModel } from '../../../types/organisations-schema'
+import { URLS } from '../../../constants/api'
+import { addMemberToGroup, removeMemberFromGroup } from '../../../services/user'
+import { toastMessages } from '../../../constants/toast-messages'
+
+export interface UpdateUserGroupModalProps {
+  editingUserGroup: GroupModel | undefined
+  setEditingUserGroup: React.Dispatch<React.SetStateAction<GroupModel | undefined>>
+  onRefetchData: () => void
+}
+
+interface UpdateUserGroupModel {
+  userId: string[]
+}
+
+type FieldType = 'groupIds'
+
+export const formFields: Record<FieldType, FormFieldInfo> = {
+  groupIds: {
+    name: 'userId',
+    label: 'Group Members',
+  },
+}
+
+export const prepareGroupOptions: (data: GroupModel[]) => SelectOption[] = data =>
+  data.map((UserGroupGroup: UserModel) => {
+    const { id, name } = UserGroupGroup
+
+    return {
+      label: name,
+      value: id,
+    } as SelectOption
+  })
+
+const addUserToGroup = async (id: string, userId: string) => {
+  const addUserRes = await addMemberToGroup({ id, userId })
+  return addUserRes
+}
+
+const removeUserFromGroup = async (id: string, userId: string) => {
+  const removeUserRes = await removeMemberFromGroup({ id, userId })
+  return removeUserRes
+}
+
+export const onHandleSubmit = (
+  handleOnClose: () => void,
+  onRefetchData: () => void,
+  mutate: () => void,
+  editingUserGroup?: GroupModel,
+  initMembers?: string[],
+) => async (params: UpdateUserGroupModel) => {
+  const id = editingUserGroup?.id || ''
+  const userId = params.userId
+  const newUserIds = difference(userId, initMembers)
+  const removeUser = difference(initMembers, userId)
```

Comment:
What type are newUserIds and removeUser? Not sure why we are using lodash here...